address-space-controller/src/main/java/io/enmasse/controller/ExportsController.java

```diff
@@ -0,0 +1,172 @@
+        }
+    }
+
+    private void exportAsConfigMap(String name, EndpointStatus endpointStatus, AddressSpace addressSpace) {
+        Map<String, String> exportMap = buildExportMap(endpointStatus);
+        ConfigMap configMap = new ConfigMapBuilder()
+                .editOrNewMetadata()
+                .withName(name)
+                .withNamespace(addressSpace.getMetadata().getNamespace())
+                .addToOwnerReferences(new OwnerReferenceBuilder()
+                        .withBlockOwnerDeletion(true)
+                        .withApiVersion(addressSpace.getApiVersion())
+                        .withController(true)
+                        .withKind(addressSpace.getKind())
+                        .withName(addressSpace.getMetadata().getName())
+                        .withUid(addressSpace.getMetadata().getUid())
+                        .build())
+                .endMetadata()
+                .addToData(exportMap)
+                .build();
+
+        ConfigMap existing = client.configMaps().inNamespace(addressSpace.getMetadata().getNamespace()).withName(name).get();
+        if (existing != null && !exportMap.equals(existing.getData())) {
+            client.configMaps().inNamespace(addressSpace.getMetadata().getNamespace()).withName(name).replace(configMap);
+        } else {
+            client.configMaps().inNamespace(addressSpace.getMetadata().getNamespace()).withName(name).createOrReplace(configMap);
+        }
+    }
+
+    private void exportAsService(String name, EndpointStatus endpointStatus, AddressSpace addressSpace) {
+        Service service = new ServiceBuilder()
+                .editOrNewMetadata()
+                .withName(name)
+                .withNamespace(addressSpace.getMetadata().getNamespace())
+                .addToOwnerReferences(new OwnerReferenceBuilder()
+                        .withBlockOwnerDeletion(true)
+                        .withApiVersion(addressSpace.getApiVersion())
+                        .withController(true)
+                        .withKind(addressSpace.getKind())
+                        .withName(addressSpace.getMetadata().getName())
+                        .withUid(addressSpace.getMetadata().getUid())
+                        .build())
+                .endMetadata()
+                .editOrNewSpec()
+                .withType("ExternalName")
+                .withExternalName(endpointStatus.getServiceHost())
+                .withPorts(ServiceHelper.toServicePortList(endpointStatus.getServicePorts()))
+                .endSpec()
+                .build();
+
+        Service existing = client.services().inNamespace(addressSpace.getMetadata().getNamespace()).withName(name).get();
+        if (existing != null &&
+                (!endpointStatus.getServiceHost().equals(existing.getSpec().getExternalName())
+                        || !ServiceHelper.toServicePortList(endpointStatus.getServicePorts()).equals(existing.getSpec().getPorts()))) {
```

Comment:
Is a list comparison what we want here?  Shouldn't this be a set?

Reply:
Fixed